addons/io_hubs_addon/components/definitions/media_frame.py

```diff
@@ -0,0 +1,148 @@
+        'display_name': 'Media Frame',
+        'category': Category.ELEMENTS,
+        'node_type': Node,
+        'panel_type': PanelType.OBJECT,
+        'icon': 'OBJECT_DATA',
+        'deps': ['networked']
+    }
+
+    bounds: FloatVectorProperty(
+        name="Bounds",
+        description="Bounding box to fit objects into when they are snapped into the media frame.",
+        unit='LENGTH',
+        subtype="XYZ_LENGTH",
+        default=(2.0, 2.0, 2.0),
+        update=update_object_bounds)
+
+    mediaType: EnumProperty(
+        name="Media Type",
+        description="Limit what type of media this frame will capture",
+        items=[("all", "All Media", "Allow any type of media."),
+               ("all-2d", "Only 2D Media", "Allow only Images, Videos, and PDFs."),
+               ("model", "Only 3D Models", "Allow only 3D models."),
+               ("image", "Only Images", "Allow only images."),
+               ("video", "Only Videos", "Allow only videos."),
+               ("pdf", "Only PDFs", "Allow only PDFs.")],
+        default="all-2d")
+
+    snapToCenter: BoolProperty(
+        name="Snap To Center",
+        description="Snap the media to the center of the media frame when capturing. If set to false the object will just remain in the place it was dorpped but still be considered \"captured\" by the media frame.",
+        default=True
+    )
+
+    @classmethod
+    def init(cls, ob):
+        ob.hubs_component_media_frame.bounds = get_gizmo_dimension(ob)
+
+    @classmethod
+    def update_gizmo(cls, ob, gizmo):
+        loc, rot, _ = ob.matrix_world.decompose()
+        scale = get_gizmo_scale(ob)
+        mat_out = Matrix.Translation(
+            loc) @ rot.normalized().to_matrix().to_4x4() @ Matrix.Diagonal(scale).to_4x4()
+        gizmo.matrix_basis = mat_out
+        gizmo.hide = not ob.visible_get()
+
+    @classmethod
+    def create_gizmo(cls, ob, gizmo_group):
+        widget = gizmo_group.gizmos.new(CustomModelGizmo.bl_idname)
+        setattr(widget, "hubs_gizmo_shape", box.SHAPE)
+        widget.setup()
+        loc, rot, _ = ob.matrix_world.decompose()
+        scale = get_gizmo_scale(ob)
+        mat_out = Matrix.Translation(
+            loc) @ rot.normalized().to_matrix().to_4x4() @ Matrix.Diagonal(scale).to_4x4()
+        widget.matrix_basis = mat_out
+        widget.use_draw_scale = False
+        widget.use_draw_modal = True
+        widget.color = (0.0, 0.0, 0.8)
+        widget.alpha = 1.0
+        widget.hide = not ob.visible_get()
+        widget.scale_basis = 1.0
+        widget.hide_select = False
+        widget.color_highlight = (0.0, 0.0, 0.8)
+        widget.alpha_highlight = 0.5
+
+        op = widget.target_set_operator("transform.resize")
+        op.constraint_axis = False, False, False
+        op.orient_type = 'LOCAL'
+        op.release_confirm = True
+
+        return widget
+
+    @classmethod
+    def migrate(cls):
+        migrate_networked(cls.get_name())
```

Comment:
Media Frames needs to migrate the bounds parameters from old files: 
```suggestion
        for ob in bpy.data.objects:
            if cls.get_name() in ob.hubs_component_list.items:
                bounds = ob.hubs_component_media_frame.bounds.copy()
                converted_bounds = Vector((bounds.x, bounds.z, bounds.y))

                if ob.type == 'MESH':
                    ob.dimensions = converted_bounds
                else:
                    ob.scale = converted_bounds / ob.empty_display_size / 2
```

However, there are some other issues with Media Frames as well.  If you have an Empty set to display as a cube, you will be unable to select it because of the gizmo.  A possible fix is to add this after `ob.scale = converted_bounds / ob.empty_display_size / 2`:
```
ob.empty_display_type = 'PLAIN_AXES'
```

but this would only fix loading old files, so maybe Media Frames should auto-switch empties to 'PLAIN_AXES' when added?

The other issue is that you have the bounds parameters linked with the scale of the object, but currently, in order for this to scale objects within the bounds of the Media Frame in Hubs the scale needs to remain at 1 (at least for empties, I haven't tested on meshes, but I would expect them to be the same).  Having the Scale at a different value will cause objects to either scale smaller or bigger than the Media Frame bounds in Hubs (For a long while I thought this was a problem, but then just tonight I realized that this is actually something I've been wanting to do for a while, but didn't connect the dots until now, lol)

Hopefully this all makes sense.

Reply:
> Media Frames needs to migrate the bounds parameters from old files

Good catch, Thanks! Looks good but I'm not sure why are you swapping the y and z components in the converted bounds. We only swap when exporting if that's enabled in the exporter.

> However, there are some other issues with Media Frames as well. If you have an Empty set to display as a cube, you will be unable to select it because of the gizmo.

I wouldn't like to force change the empty type as that is a user choice and I'd prefer to respect that. I've tried to scale the gizmo so it's a bit offsetted from the object but there seems to be already a selection offset so it doesn't really work well. So not sure what's the best option...

> The other issue is that you have the bounds parameters linked with the scale of the object, but currently, in order for this to scale objects within the bounds of the Media Frame in Hubs the scale needs to remain at 1 (at least for empties, I haven't tested on meshes, but I would expect them to be the same). Having the Scale at a different value will cause objects to either scale smaller or bigger than the Media Frame bounds in Hubs (For a long while I thought this was a problem, but then just tonight I realized that this is actually something I've been wanting to do for a while, but didn't connect the dots until now, lol)

Yep, I totally forgot about that. Thinking again about this, it seems that keeping the object transform separated from the gizmo transform seems the most reasonable thing. I'm going to make them independent and that should fix this issue and make it work in a more reasonable way.

Reply:
I've updated to separate the object and the gizmo transforms in the media frame component. I think makes sense for this component.

Reply:
> > Media Frames needs to migrate the bounds parameters from old files
> 
> Good catch, Thanks! Looks good but I'm not sure why are you swapping the y and z components in the converted bounds. We only swap when exporting if that's enabled in the exporter.

Because the old version of the exporter didn't swap them on export, ever, so in order to have them show up correctly now they need to be swapped when migrated.

> I wouldn't like to force change the empty type as that is a user choice and I'd prefer to respect that. I've tried to scale the gizmo so it's a bit offsetted from the object but there seems to be already a selection offset so it doesn't really work well. So not sure what's the best option...

Yeah, I'm not really sure what the best solution is.  With the recent changes to the gizmo actions it's now easier to select them, but I always worry when I click on the gizmo that I'll accidentally change something, rather than just selecting it, but it's probably good enough for now.

> I've updated to separate the object and the gizmo transforms in the media frame component. I think makes sense for this component. We are now also resetting the object scale to 1 when exporting like Spoke does otherwise we have the issues that you mentioned.

I see that.  Resetting the object scale does avoid the bug (I was mistaken before, it's not the feature I was hoping it was), and is possibly the best solution, but I dislike changing stuff auto-magically behind the scenes on users.  Could you add a warning to the component, when the object scale isn't 1, that the scale will be reset when exporting?

Reply:
> Because the old version of the exporter didn't swap them on export, ever, so in order to have them show up correctly now they need to be swapped when migrated.

This `migrate` method is used when loading the component properties from the blend file. I think we will need this when importing GLTF files exported with the previous addon versions but not in this case.

> I'm not so sure this is the right thing to do anymore. The way you're applying the scale alters the transforms of children (and the children's children, etc.), and won't work on multi-user mesh objects. While I'm sure a solution can be found to prevent the alteration of children, and multi-user mesh objects could be converted into single user ones and then converted back to multi-user in the post-export hook, I'm starting to wonder if it would be better to just display a warning on the component when the object doesn't have a scale of 1,1,1?

In Spoke we have full control over the component gizmo but in Blender we depend on the user's chosen parent object and if the scale it's not [1,1,1] either the `media-frame` gizmo or the ammo shape won't be correct in Hubs. As long as we can't create custom Blender objects from Python there is not a perfect solution for this.

You made good points, I agree that resetting the scale might add confusion in case of complex hierarchies (also I didn't really think on the multi user case)... I'll remove applying the scale and just show the warning for now. I don't like that there are so many uncertainties for the user, it's not great for usability, I don't think a regular user will really see the warning unless it's looking for it but I think this will work for now